### PR TITLE
Document operation with the query scheduler.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This patch adds documentation on the query scheduler supported added via #4071.

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/operations/scalability.md [hunk 1]
```diff
@@ -12,3 +12,11 @@ partitioned by role (ingester, distributor, querier) rather than a single Loki
 process. Grafana Labs' [production setup](https://github.com/grafana/loki/blob/master/production/ksonnet/loki)
 contains `.libsonnet` files that demonstrates configuring separate components
 and scaling for resource usage.
+
+## Separate Query Scheduler
```
Comment (jeschkies) [inline docs/sources/operations/scalability.md:16]: This is the doc.

File: docs/sources/operations/scalability.md [hunk 2]
```diff
@@ -12,3 +12,11 @@ partitioned by role (ingester, distributor, querier) rather than a single Loki
...
+
+The Query frontend has an in-memory queue that can be moved out into a separate process similar to the [Cortex Query Scheduler](https://cortexmetrics.io/docs/operations/scaling-query-frontend/#query-scheduler). This allows running multiple query frontends.
+
+In order to run with the Query Scheduler, the frontend needs to be passed the scheduler's address via `-frontend.scheduler-address` and the querier processes needs to be started with `-querier.scheduler-address` set to the same address. It is not valid to set `-querier.frontend-address` and `-querier.frontend-address`. Both options can also be set via the [configuration file](../configuration).
```
Comment (KMiller-Grafana) [inline docs/sources/operations/scalability.md:20]: Please revise the second sentence.  I don't know what it intends to say, but you have the same `-querier.frontend-address`  twice in the sentence.  If you're trying to specify that you cannot set both items in the configuration, then I'd prefer to see wording similar to:
Specify only one of `first-one` and `second-one` when invoking the <fill in here>; it is not valid to specify both.

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

File: docs/sources/operations/scalability.md (hunk 2 above)
Comment (jeschkies) [reply docs/sources/operations/scalability.md:20]: Thanks. This is a typo. IT's not valid to specify `querier.frontend-address` and `querier.scheduler-address`, I'll clean this up.

File: docs/sources/operations/scalability.md [hunk 3]
```diff
@@ -12,3 +12,13 @@ partitioned by role (ingester, distributor, querier) rather than a single Loki
...
 contains `.libsonnet` files that demonstrates configuring separate components
 and scaling for resource usage.
+
+## Separate Query Scheduler
+
+The Query frontend has an in-memory queue that can be moved out into a separate process similar to the [Cortex Query Scheduler](https://cortexmetrics.io/docs/operations/scaling-query-frontend/#query-scheduler). This allows running multiple query frontends.
+
+In order to run with the Query Scheduler, the frontend needs to be passed the scheduler's address via `-frontend.scheduler-address` and the querier processes needs to be started with `-querier.scheduler-address` set to the same address. Both options can also be defined via the [configuration file](../configuration).
+
+Please note, that it is not valid to start the querier with a configured frontend and scheduler address.
```
Comment (KMiller-Grafana) [inline docs/sources/operations/scalability.md:22]: This sentence still needs a little work.  How about:

```suggestion
It is not valid to start the querier with both a configured frontend and a scheduler address. 
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

**Review by KMiller-Grafana — APPROVED** [review]
Thanks for the changes.  Docs LGTM now.

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**